main.cpp

```diff
@@ -5,45 +5,46 @@
 Create a branch named Part2
 
  References
- 
- 
- 1) convert the pointer usage (except for 'const char*') to reference types or 
+
+
+ 1) convert the pointer usage (except for 'const char*') to reference types or
     const reference types **>>> WHERE POSSIBLE <<<**
     Not every pointer can be converted.
-        hint: There is no reference equivalent to nullptr.  
+        hint: There is no reference equivalent to nullptr.
         if a pointer (including nullptr) is being returned anywhere, don't try to convert it to a reference and don't change the return type either.
 
     You have to ask yourself if each pointer can be converted to a (const) reference.
     Think carefully when making your changes.
 
- 2) revise the 'else' statement in main() that handles when `smaller` is a nullptr. 
- there is only one reason for `compare` to return nullptr. 
+ 2) revise the 'else' statement in main() that handles when `smaller` is a nullptr.
+ there is only one reason for `compare` to return nullptr.
  Update this std::cout expression to specify why nullptr was returned.
- 
+
  3) After you finish, click the [run] button.  Clear up any errors or warnings as best you can.
- 
+
  */
 
 
 #include <iostream>
 #include <string>
-struct T 
+
+struct T
 {
     int distance;
     std::string name;
-    T(int v, const char* letterName) 
+    T(int v, const char* letterName)
     {
         if (letterName == nullptr) return;
         distance = v;             // Implicit "this"
         this->name = *letterName; // Explicit "this"
     }
 };
 
-struct A 
+struct A
 {
-    T* compare(T* a, T* b) 
+    T* compare(T* a, T* b) // Should not be converted to reference due to nullptr comparison
```

Comment:
You can convert the arguments.
you can't convert the return type.
- [ ] revise